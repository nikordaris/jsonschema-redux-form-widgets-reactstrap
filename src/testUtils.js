import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { reduxForm, reducer as formReducer } from 'redux-form';
import { createStore, combineReducers } from 'redux';
import { shallow } from 'enzyme';

export const matchSnapshotWithReduxForm = (options, TestComponent) => () => {
  const rootReducers = combineReducers({ form: formReducer });
  const store = createStore(rootReducers);
  const WrappedComponent = reduxForm({ form: 'MyForm' })(TestComponent);
  const tree = renderer
    .create(
      <Provider store={store}>
        <WrappedComponent {...options} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
};

export const matchSnapshot = (options, TestComponent) => () => {
  const tree = renderer.create(<TestComponent {...options} />).toJSON();
  expect(tree).toMatchSnapshot();
};

export const matchSnapshotShallow = (options, TestComponent) => () => {
  const wrapper = shallow(<TestComponent {...options} />);

  expect(wrapper.getNodes()).toMatchSnapshot();
};
