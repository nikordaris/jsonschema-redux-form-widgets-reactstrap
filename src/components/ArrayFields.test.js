import React from 'react';
import { UniformedArrayCard, Text, CardItem } from '../index';
import { matchSnapshotWithReduxForm } from '../testUtils';

describe('Render ArrayFields', () => {
  it(
    'Should render UniformedArrayCard',
    matchSnapshotWithReduxForm(
      {
        schemaVis: {
          prefix: 'meta.vis',
          schema: {
            id: 'UniformedArrayCard',
            title: 'Uniformed Array Card',
            type: 'array',
            items: {
              id: 'Bar',
              title: 'Bar',
              type: 'object',
              properties: {
                foo: { type: 'string', meta: { vis: { component: 'Text' } } }
              },
              meta: { vis: { component: 'CardItem' } }
            }
          },
          components: { Text, UniformedArrayCard, CardItem }
        },
        name: 'Foo',
        form: 'MyForm'
      },
      UniformedArrayCard
    )
  );
});
