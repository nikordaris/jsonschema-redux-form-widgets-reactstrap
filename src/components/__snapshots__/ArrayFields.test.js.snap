// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ArrayFields Should render UniformedArrayCard 1`] = `
<div
  className="container-0-0 card"
>
  <div
    className="header-0-1 card-header"
  >
    <div
      className="headerTitle-0-3"
    >
      Uniformed Array Card
    </div>
    <div
      className="addButton-0-2"
    >
      <button
        className="btn btn-primary btn-sm"
        onClick={[Function]}
        type="button"
      >
        Add
      </button>
    </div>
  </div>
  <div
    className="card-block"
  />
</div>
`;
